### DISCUSSION TITLE
floating point exception issue

### DISCUSSION THREAD

**Discussion by SmithSimmon** [NCAR/hrldas#211] [Q&A] [answered]
Hi, everyone! I ran a simulation from 2015 to 2100 using CMIP6 data. The model crashed in November 11, 2030. I tried to skip this day and continue running the model, but after running for a few years, the model encountered the same error. the error is mainly about surfacEenergyFluxMod and ResistancecanopystomataBallBerryMod.
Here is error message:
<img width="657" alt="903bbbddfcba89c7b7806857bcd7106" src="https://github.com/user-attachments/assets/ef6748c3-b8dc-49c6-983a-a46f76758e88"> 
year 2030
![image](https://github.com/user-attachments/assets/560d8263-510f-451a-872b-25a64cfb8af5)
year 2036
the command line from error message in SurfaceEnergyFluxMod f90 file: 
TempTmp = TempUnitConv(TemperatureGrdBare)
I don't know why I am meeting a floating-point issue. I have checked the temperature forcing data and radiation forcing data, and the values seem to be within a reasonable range. Could you please give me some advice?
By the way, is there any method to know the exact grid where the problem occurs？

**Comment by cenlinhe** [comment, marked as answer]
What model time step are you using? If you are using an hourly timestep (NOAH_TIMESTEP = 3600), then please reduce it to half-hourly (NOAH_TIMESTEP = 1800). The issue might be caused by the inaccurate numerical solution under a large timestep.
If you want to know the exact grid, you could add printout code in the NoahmpDriverMain module for the I, J point.

**Reply by SmithSimmon** [reply]
thanks for u help! I have solved this issue.

**Comment by feifeiyu1225** [comment]
May I ask how you solved this problem? I'm facing a similar one.
![微信图片_20250415074035](https://github.com/user-attachments/assets/413ffb16-dc96-449c-95b6-20ef3283e150)
